programs/ziti-edge-tunnel/instance.c

```diff
@@ -34,14 +36,36 @@ tunnel_identity *find_tunnel_identity(const char* identifier) {
     }
 }
 
-tunnel_identity *get_tunnel_identity(char* identifier) {
+tunnel_identity *create_or_get_tunnel_identity(char* identifier, char* filename) {
     tunnel_identity *id = find_tunnel_identity(identifier);
 
     if (id != NULL) {
+        if (filename != NULL) {
+            id->Status = strdup(instance_status_name(instance_status_ok));
+        }
         return id;
     } else {
         tunnel_identity *tnl_id = calloc(1, sizeof(struct tunnel_identity_s));
         tnl_id->Identifier = strdup(identifier);
+        if (filename != NULL) {
+            char* extension = strstr(filename, ".json");
+
+            if (extension != NULL) {
+                int length = (int) (extension - filename);
+                tnl_id->FingerPrint = calloc(length + 1, sizeof(char));
```

Comment:
if `FingerPrint` just filename without `.json`? I don't understand the need for it

Reply:
Also it is not set if filename does not end with `.json`. This seems flaky

Reply:
Fingerprint is the identifier that WDE UI / go tunnel code uses currently. We can remove it later, once all the references in the UI is changed to the Identifier variable. That may take time. So I thought, the fastest way to get this integrated with WDE UI, is to add Fingerprint variable as well in to the event messages

Reply:
fixed the issue with the null fingerprint value, when the file name does not end with .json